examples/tests/valid/json.w

```diff
@@ -52,7 +52,7 @@ someJson.set("x", 111);
 assert(someJson.get("x") == 111);
 
 // assign Map to Json
-let x: Json = {cool: "beans"};
+let x: Json = {cool => "beans"};
```

Comment:
This is Json, not Map, right? This change shouldn't be needed (`Json { a: b }` is still the syntax used in the spec)

Reply:
@Chriscbr so both would be valid

```
let x: Json = {cool: "beans"}; // Json map
OR
let x: Json = {cool => "beans"}; // non-json map
```

I only made the change to show that we can still assign map elements into Json. Like we do with arrays `Json [1, 2, 3]`

Reply:
Actually this is a current issue where everything is a subtype of Json. Im going to add a subsequent PR with plans to remove that. 

For now we can see the desired spec behavior here: https://github.com/winglang/wing/pull/3018/commits/c295f9ac0a4ff33768cf4e1dc338f98db2bdf89e#diff-25a9b7a88f48644f959355fc63e91f28fbe6751861da3ec0b689da4d37c141efR133-R136

Reply:
Got it. I'll be honest, allowing both `Json { a: "b" }` and `Json { a => "b" }` as two ways to construct the same object feels a bit odd (like we've made some mistake in our syntax design somewhere) -- but maybe it will be uncommon enough to happen in practice that it won't be an issue. In any case, I'm down to try this new syntax

Reply:
Yea I actually really dislike this when working within Structs that contain maps. It feels wonky

```js
struct A {
	foo: str;
	bar: Map<str>;
}

let a = A {
	foo: "foofoo",
	bar: {
		x => "a"
	}
}
```

Reply:
> Got it. I'll be honest, allowing both `Json { a: "b" }` and `Json { a => "b" }`

I think the 2nd option above isn't legal in our syntax:
```wing
let a = Map<str> { a => "b" }; // a map
let b =  { a => "b" }; // a map initialized with a map without the explicit type
let c: Json =  { a => "b" }; // a Json initialized with a map
let d = Json { a: "b" }; // a Json initialized with a Json object
let e = { a: "b" }; // a Json initialized with a Json object without the explicit type
let f: Map<str> = { a: "b" }; // complication error: can't initialize a map with a Json literal
```

Reply:
It makes sense to me that `Map` and `Set` won't be implicitly casted to Json (because they don't have a reversible representation). But primitive types and arrays should be implicitly casted.

Reply:
I made some updates here to not allow maps and sets to be valid json. I also added distinct ExprKind for JsonMapLiteral so we arent overloading the use of MapLiteral in typechecking and jsification to deal with both normal maps and Json. Keeps the responsibilities separate I think.

Reply:
A winglang map becomes a JavaScript `Map` object. When JSON.stringified it becomes a "regular" JavaScript object, which means that when parsed back to JavaScript it won't be a Map anymore:

```js
const y = new Map();
const x = JSON.parse(JSON.stringify(y));
assert(y instanceof Map); // ok!
assert(x instanceof Map); // boom!
```

Reply:
@eladb I'm afriad you're mistaken - Winglang maps (as of this comment) are code-generated as plain JavaScript objects (not [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) objects) - you can try compiling a program in the playground to validate this. This behavior is necessary to interoperate with the CDKTF / AWS CDK libraries, where any type declared in TypeScript as `Record<string, T>` or `{ [key: string]: T }` is treated as a map in the JSII model.

Reply:
Discussed with @eladb -he's ok with the current state. Resolving.